connector_magento_product_catalog/models/product/listener.py

```diff
@@ -47,6 +47,8 @@ class MagentoProductPricelistItemUpdateListener(Component):
     _apply_on = ['product.pricelist.item']
 
     def update_products(self, record):
+        if self.backend_id.product_synchro_strategy == 'magento_first': 
+                return
```

Comment:
We should generally add a force context option here - so that the export is possible also if magento_first is set. For example for the Direct Export Button on the binding record. If we don't add it - then we should disable the export buttons on the binding records - because they would have no affect...

Reply:
Ok,

I agree that we should express more clearly the available options, Yet I'm not sure that our immediate proposals are accurate. 

What i would like to clearly present for the user is the fact that its backend is either set for import (classical connector_magento module) or for export (first option + all the catalogue functions).

If you share my point of view I would propose to hide the export / Import button regarding the backend configuration.

Let me know

Reply:
I think the original concept was an other concept...


The exporter function does call the _should_import function - this function does check if the odoo last modification date is greater than the magento last modification date. If the magento date is greater - then it will return true.

If it does return true - then the _delay_import will get called. In the _delay_import now we should check the sync strategy - if it is set to magento_first - then run immediate the import - and then the export. If set to odoo first - then run the export first - and delay the import.

So theoretically it should never check the sync strategy in the listener functions - because the export job should get run - and the export job then does the check.

Reply:
ok, I understand and I also remember.

Reply:
@wpichler , could you just remind me why the _should_import method is named this way as it is only used in export ?

Reply:
the _should_import function should decide if the record should get imported before it gets exported. If _should_import does return true - then the record on magento side does has a change that we don't have on odoo side. With the selected strategy we have to decide if we do import the record before the export (so magento data does overwrite odoo data = magento_first) or if we do export the record - and then import the record from magento (=odoo_first)

Reply:
As far as i can see in the code - is it currently more a mess than consistent implemented. So this can get refactored completly

Reply:
ok, thanks

Reply:
BTW, maybe the field strategy is not properly named.
My idea was to either import from magento product data or export product data. 
Did you understood this field the same way? Do I need to refactor as well ?

Reply:
I did understand the field as priority field - it does specify if magento data has priority over odoo data or the other way round. A field for export or not is not needed - because the export functionality is in a separate module - so if you don't install export module - then you want have the export. I think we should refactor it to be clearly a priority strategy field.